### PR TITLE
ARROW-8621: [Release] Add post release step to add tags for Go versioning

### PR DESCRIPTION
**Pull request by zeroshade**
Adding a new post script for adding the tags needed for pkg.go.dev to recognize the package versioning for the Golang modules for arrow and parquet

### CODE REVIEW

**Comment by zeroshade** [comment]
Also, specifically I wanted to get this in in time for the 6.0.0 release to hopefully get that versioned properly in Go via the tags

File: dev/release/post-13-go.sh [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+version=$1
+go_arrow_tag="go/arrow/v${version}"
+go_parquet_tag="go/parquet/v${version}"
+
+git tag "${go_arrow_tag}"
+git tag "${go_parquet_tag}"
+
+git push origin "${go_arrow_tag}"
+git push origin "${go_parquet_tag}"
```
Comment (kou) [inline dev/release/post-13-go.sh:37]: Could you use `apache` as remote name? We assume `apache` in https://cwiki.apache.org/confluence/display/ARROW/Release+Management+Guide .

```suggestion
git push apache "${go_arrow_tag}"
git push apache "${go_parquet_tag}"
```

File: dev/release/post-13-go.sh [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    echo "Usage: $0 <version>"
+    exit
+fi
+
+version=$1
+go_arrow_tag="go/arrow/v${version}"
+go_parquet_tag="go/parquet/v${version}"
+
+git tag "${go_arrow_tag}"
+git tag "${go_parquet_tag}"
```
Comment (kou) [inline dev/release/post-13-go.sh:34]: Could you ensure using `apache-arrow-${version}` as tagged version?

```suggestion
version=$1
version_tag="apache-arrow-${version}"
go_arrow_tag="go/arrow/v${version}"
go_parquet_tag="go/parquet/v${version}"

git tag "${go_arrow_tag}" "${version_tag}"
git tag "${go_parquet_tag}" "${version_tag}"
```

**Review by kou — COMMENTED** [review]
I added all permissions on https://cwiki.apache.org/confluence/display/ARROW/ to you.

`post-13-go.sh` is OK. `01-prepare.sh` isn't suitable because it's used in RC release process.

Could you update this pull request's description to be usable for commit message?

**Comment by zeroshade** [comment]
@kou Updated the description and your suggestions. I'll update the confluence

**Comment by zeroshade** [comment]
@kou Confluence updated, please let me know if there's anything else i need to do (or if there's any issue with my added task of "Updating tags for Go modules" to the confluence.)

**Comment by kou** [comment]
Thanks. I've confirmed.

https://cwiki.apache.org/confluence/display/ARROW/Release+Management+Guide

> The script assumes the remote name is apache  and does not push the tags itself, that needs to be done manually after running it.

The script pushes the tags automatically by the last 2 lines. Which behavior do you expect?

**Review by kou — APPROVED** [review]
+1

I'm OK with pushing tags automatically.

**Pull request closed without merging**